nautobot/extras/tests/test_dynamicgroups.py

```diff
@@ -828,6 +840,107 @@ def test_delete(self):
         self.nested_child.parents.clear()
         self.nested_child.delete()
 
+    def test_filter_relationships(self):
+        """Test that relationships can be used in filters."""
```

Comment:
This looks like a good set of test cases on the model side. Should we also have tests on the forms or UI side? (I note that `nautobot.extras.tests.test_views.DynamicGroupTestCase` is only testing the initial "create" form data, seems like we should have a test there that's submitting a filter string or two to make sure it's accepted...?)

Reply:
Do you mean tests each for relationships, custom fields, and searches? Because oof...

The basic view tests aren't even testing nested dynamic groups either. There's a lot of gap in coverage here.

Reply:
I was thinking just a basic set of tests of "can I submit a form to update an existing dynamicgroup to include a filter? a filter containing a relationship/custom field?"